src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -54,6 +54,12 @@ private enum TimeZoneInfoResult
         private const string UtcId = "UTC";
         private const string LocalId = "Local";
 
+#if TARGET_WINDOWS
+        // This field is only used in TimeZoneInfo.Win32.cs, but needs to be defined here to ensure it is initialized
+        // before the call to CreateUtcTimeZone during static field initialization of s_utcTimeZone.
+        private static readonly Dictionary<string, string> s_FileMuiPathCache = new();
```

Comment:
@jkotas is there a better way to force this static initialization without having this declared in this source file? I want to avoid using #if as possible.

Reply:
FYI, [here's one of the failures](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-52992-merge-a0b1522db1ad44bf8d/Common.Tests/console.fcec7b84.log?sv=2019-07-07&se=2021-06-09T01%3A27%3A06Z&sr=c&sp=rl&sig=gFev4J9dBOxsoHYhwUCOnOHKnk5162uW1S1nr3x4sAQ%3D) from before I moved it.  My understanding is that static type initialization across partial class files has an indeterminate order, but within a given file the order is consistent.

Reply:
No risk more than I wants to avoid using `#if` and want to have this declaration back to Win32 file which is the right place to declare it.

Reply:
Just wondering - should this dictionary be part of the cached data? Then this ordering problem goes away.

If it really needs to be a static field, you can move both fields to Win32.cs file and duplicate the one static field in Unix.cs field. It would avoid the ifdef at the cost of small code duplication.

Reply:
Nah, it shouldn't be part of CachedData.  It's a cache, but it has different concerns and lifetime than the time zones under CachedData.

I took your idea of moving/duplicating the static `s_utcTimeZone` field.  The net result is the same as before, but without any `#if`'s.